src/components/Transaction/TransactionStats.jsx

```diff
@@ -0,0 +1,174 @@
+
+  const handleStartChange = (event) => {
+    setStartDate(event.target.value);
+  };
+
+  const handleEndChange = (event) => {
+    setEndDate(event.target.value);
+  };
+
+  useEffect(() => {
+    API.instance.get('/transaction').then((res) => {
+      setTransactions(res.data);
+      setLoading(false);
+    });
+  }, []);
+
+  console.log(transactions);
+
+  const salesInRange = dataUtil.getSalesInRange(
+    transactions,
+    startDate,
+    endDate,
+  );
+
+  const incomeByDay = dataUtil.getIncomeByDay(salesInRange);
+
+  const purchasesInRange = dataUtil.getPurchasesInRange(
+    transactions,
+    startDate,
+    endDate,
+  );
+
+  const restocksInRange = dataUtil.getRestocksInRange(
+    transactions,
+    startDate,
+    endDate,
+  );
+
+  const renderPlot = () => {
+    if (loading) return <ReactLoading color="#e26d5c" />;
+    if (incomeByDay.length > 0)
+      return (
+        <div>
+          <h3>Income vs. Date</h3>
+          <LineChart width={600} height={400} data={incomeByDay}>
+            <CartesianGrid strokeDasharray="3 3" />
+            <XAxis label="Date" dataKey="date" interval="preserveEnd" />
+            <YAxis label="$" interval="preserveEnd" />
+            <Line
+              type="monotone"
+              dataKey="income"
+              stroke="#8884d8"
+              activeDot={{ r: 8 }}
+            />
+          </LineChart>
+        </div>
+      );
+  };
+
+  const dateInputs = () => {
+    if (loading) return <ReactLoading color="#e26d5c" />;
+    return (
+      <div className="container">
+        <FormGroup bssize="large">
+          <FormLabel>Start Date</FormLabel>
+          <FormControl
+            name="startDate"
+            type="date"
+            value={startDate}
+            onChange={handleStartChange}
+          />
+        </FormGroup>
+
+        <FormGroup bssize="large">
+          <FormLabel>End Date</FormLabel>
+          <FormControl
+            name="startDate"
+            type="date"
+            value={endDate}
+            onChange={handleEndChange}
+          />
+        </FormGroup>
+      </div>
+    );
+  };
+
+  const filteredSales = () => {
+    if (loading) return <ReactLoading color="#e26d5c" />;
+    return (
+      <>
+        <h3>Sales Transactions in Date Range</h3>
+        {salesInRange.map((transaction) => (
+          <TransactionListItem
+            transaction={transaction}
+            key={transaction.transactionID}
+          />
+        ))}
+      </>
+    );
+  };
+
+  const filteredPurchases = () => {
+    if (loading) return <ReactLoading color="#e26d5c" />;
+    return (
+      <>
+        <h3>Purchase Transactions in Date Range</h3>
+        {purchasesInRange.map((transaction) => (
+          <TransactionListItem
+            transaction={transaction}
+            key={transaction.transactionID}
+          />
+        ))}
+      </>
+    );
+  };
+
+  const filteredRestocks = () => {
+    if (loading) return <ReactLoading color="#e26d5c" />;
```

Comment:
Should we just have one big loading indicator instead of many?

Reply:
yea fixed.